### ISSUE TITLE
[qemu] q35: PCI bus rescan code is messing up PCIe hotplug

### ISSUE THREAD

**Issue by devimc** [kata-containers/agent#781]
see https://github.com/kata-containers/runtime/issues/2664

**Comment by dgibson** [comment]
This issue covers a number of separate bugs.  The one that's most relevant (and not covered by other more specific issues) is the problems caused by PCI rescans colliding with hotplug operations.  Can we retitle it to reflect that?

**Comment by dgibson** [comment]
As @jodh-intel pointed out in #782, removing the rescan has the side effect that we don't wait until the hotplugged device is ready before starting the container, which breaks the CI (probably amongst other things).

I'm submitting a new PR which includes @devimc's patch to remove the rescan along with extra logic to have the agent explicitly wait for the VFIO hotplug to complete.

**Comment by dgibson** [comment]
I'm no longer planning to try to get this finished for Kata1, I'll be pursuing it with Kata2 instead.
